streaming/base/partition/orig.py

```diff
@@ -80,37 +81,53 @@ def get_partitions_orig(num_samples: int,
-    #
-    # row_offsets: (canonical nodes, 1).
-    row_offsets = np.arange(num_canonical_nodes) * padding
-    row_offsets = np.expand_dims(row_offsets, 1)
-    ids -= row_offsets
-
-    # Reconfigure where each row starts iterating for irregular-sized rows.
-    #
-    # row_starts: (canonical nodes, 1).
-    row_starts = np.arange(num_canonical_nodes) * num_samples // num_canonical_nodes
-    row_starts = np.expand_dims(row_starts, 1)
-    ids += row_starts - ids[:, :1]
-
-    # For short rows (length not evenly divisible), repeat the last ID to get even length.
-    #
-    # row_stops: (canonical nodes, 1).
-    row_stops = np.arange(1, 1 + num_canonical_nodes) * num_samples // num_canonical_nodes
-    row_stops = np.expand_dims(row_stops, 1)
-    are_rows_short = row_stops - row_starts < samples_per_canonical_node
-    ids[:, samples_per_canonical_node - 1:samples_per_canonical_node] -= are_rows_short
-
-    # If padding we needed, repeat samples to populate it.
-    if padding:
-        ids[:, -padding:] = ids[:, -padding - node_ratio + 1 - padding:-padding - node_ratio + 1]
+    if num_samples > num_canonical_nodes:
+        # Create the initial sample ID matrix.
+        #
+        # ids: (canonical nodes, padded samples per canonical node).
+        ids = np.arange(num_canonical_nodes * padded_samples_per_canonical_node, dtype=np.int64)
+        ids = ids.reshape(num_canonical_nodes, padded_samples_per_canonical_node)
+
+        # Adjust row offsets to ignore the padding.
+        #
+        # row_offsets: (canonical nodes, 1).
+        row_offsets = np.arange(num_canonical_nodes) * padding
+        row_offsets = np.expand_dims(row_offsets, 1)
+        ids -= row_offsets
+
+        # Reconfigure where each row starts iterating for irregular-sized rows.
+        #
+        # row_starts: (canonical nodes, 1).
+        row_starts = np.arange(num_canonical_nodes) * num_samples // num_canonical_nodes
+        row_starts = np.expand_dims(row_starts, 1)
+        ids += row_starts - ids[:, :1]
+
+        # For short rows (length not evenly divisible), repeat the last ID to get even length.
+        #
+        # row_stops: (canonical nodes, 1).
+        row_stops = np.arange(1, 1 + num_canonical_nodes) * num_samples // num_canonical_nodes
+        row_stops = np.expand_dims(row_stops, 1)
+        are_rows_short = row_stops - row_starts < samples_per_canonical_node
+        ids[:, samples_per_canonical_node - 1:samples_per_canonical_node] -= are_rows_short
+
+        # If padding we needed, repeat samples to populate it.
+        if padding:
+            ids[:, -padding:] = ids[:,
+                                    -padding - node_ratio + 1 - padding:-padding - node_ratio + 1]
+    else:
+        # Num samples is less than the number of canonical nodes. The dataset is very small, and
+        # samples will be repeated extensively.
+        warnings.warn(f'Attempting to partition {num_samples} samples over {num_canonical_nodes} \
```

Comment:
Is there any recommendation we need to provide here? Such as lowering NCN or something else?

Reply:
yeah recommendation here is to lower NCN, and check if the dataset has the expected number of samples. added in.